stdlib/public/Darwin/Accelerate/vImage_Buffer.swift

```diff
@@ -0,0 +1,348 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Accelerate
+
+//===----------------------------------------------------------------------===//
+//
+//  vImage_Buffer
+//
+//===----------------------------------------------------------------------===//
+
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+extension vImage_Buffer {
+    
+    /// Returns the size, in pixels, of a vImage buffer.
+    public var size: CGSize {
+        var mutableSelf = self
+        return vImageBuffer_GetSize(&mutableSelf)
+    }
+    
+    //===----------------------------------------------------------------------===//
+    
+    /// Initializes a vImage buffer of a specified size without memory allocation,
+    /// reporting the preferred alignment.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter alignment: The preferred alignment is written to this parameter.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
+                 bitsPerPixel: UInt32,
+                 alignment: inout Int) {
+        
+        self.init()
+        
+        if size.width < 0 || size.height < 0 {
+            return nil
+        }
+        let width = vImagePixelCount(size.width)
+        let height = vImagePixelCount(size.height)
+        
+        let error = vImageBuffer_Init(&self,
+                                      height,
+                                      width,
+                                      bitsPerPixel,
+                                      vImage_Flags(kvImageNoAllocate))
+        
+        if error < kvImageNoError {
+            return nil
+        } else {
+            alignment = error
+        }
+    }
+    
+    /// Initializes a vImage buffer of a specified size, reporting any errors.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter error: Overwritten with the error code if the operation failed.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
```

Comment:
Why are we taking `CGSize` params for these functions? `CGSize` uses `CGFloat` to represent `width` and `height`, which causes problems due to rounding on 32b platforms (watchOS). Unless you have specific use cases in mind, it seems vastly preferable to use `Int` `width` and `height` parameters.

Reply:
My thinking was that the native `width` and `height` properties of a `vImage_Buffer` are `vImagePixelCount` (`UInt`) - it seemed a bit disjointed to have an integer parameter on the initializer, but the property exposed as a different type. Another point was that all existing vImage functions have `height` preceding `width` and that using `CGSize` might assist in avoiding errors with my `width` preceding `height`. 

That said, other image classes do use integer width and height, so I'll update accordingly.